### PR TITLE
FAQ link fixes

### PR DESCRIPTION
**Pull request by dennisreimann**

### CODE REVIEW

**Comment by pavlenex** [comment]
@dennisreimann  All of these links become broken on GitHub then 😿

File: docs/FAQ/FAQ-Synchronization.md [hunk 1]
```diff
@@ -185,13 +185,13 @@ tmpfs           2.0G     0  2.0G   0% /sys/fs/cgroup
 /dev/sdb1       7.8G   18M  7.4G   1% /mnt
 ```
 
-[Choose the docker fragment](https://github.com/btcpayserver/btcpayserver-docker#generated-docker-compose-) for the amount of storage you aim to keep.
+[Choose the docker fragment](https://github.com/btcpayserver/btcpayserver-docker/blob/master/README.md#generated-docker-compose) for the amount of storage you aim to keep.
 
-Then [prune your node](https://github.com/btcpayserver/btcpayserver-docker#how-i-can-prune-my-nodes).
+Then [prune your node](https://github.com/btcpayserver/btcpayserver-docker/blob/master/README.md#how-i-can-prune-my-node-s).
```
Comment (britttttk) [inline docs/FAQ/FAQ-Synchronization.md:190]:
```suggestion
Then [prune your node](https://github.com/btcpayserver/btcpayserver-docker/blob/master/README.md#how-i-can-prune-my-nodes).
```

Comment (dennisreimann) [reply docs/FAQ/FAQ-Synchronization.md:190]: Thanks for pointing that out. Actually this is a bit more complicated: The Vuepress slug function seems to diverge from what GitHub's markdown parser generates … will have to dig a bit deeper and see how we can make it work on GH as well as on the doc site.

Comment (dennisreimann) [reply docs/FAQ/FAQ-Synchronization.md]: Hope I fully addressed it with 60a5ccb.

**Comment by dennisreimann** [comment]
Ok, there were more cases than just in the FAQ, often related to Vuepress' slug output being different than that on GitHub. 

As some more cases of broken links were reported on Mattermost today I've also added a link checker, that scans the site for broken links. Weirdly it reports some external links as broken, even though they work – sometimes it's because the HTTP status code isn't appropriate.

There are also PRs in the [Docker](https://github.com/btcpayserver/btcpayserver-docker/pull/321) and [Transmuter](https://github.com/btcpayserver/btcTransmuter/pull/64) repos, to address broken links in their Markdown files.

I'd say we get this merged and then see how we can improve on the link checks. For now it runs as the last step in the pipeline, but it doesn't cancel the deployment or such. We'd definitely have to add exclude rules for the external links that work but report an error.

**Comment by dennisreimann** [comment]
Rebased to integrate @pavlenex' wallet faq improvements :)

**Pull request merged**